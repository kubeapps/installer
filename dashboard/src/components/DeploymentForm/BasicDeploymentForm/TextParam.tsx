import * as React from "react";
import { IBasicFormParam } from "shared/types";

export interface IStringParamProps {
  id: string;
  name: string;
  label: string;
  inputType?: string;
  param: IBasicFormParam;
  handleBasicFormParamChange: (
    name: string,
    p: IBasicFormParam,
  ) => (e: React.FormEvent<HTMLInputElement>) => void;
}

class TextParam extends React.Component<IStringParamProps> {
  public render() {
    const { id, name, param, label, inputType } = this.props;
    return (
      <div>
        <label htmlFor={id}>
          {label}
          {param.description && (
            <>
              <br />
              <span className="description">{param.description}</span>
            </>
          )}
          <input
            id={id}
            onChange={this.props.handleBasicFormParamChange(name, param)}
            defaultValue={param.value}
            type={inputType ? inputType : "text"}
          />
        </label>
      </div>
    );
  }
}

export default TextParam;
