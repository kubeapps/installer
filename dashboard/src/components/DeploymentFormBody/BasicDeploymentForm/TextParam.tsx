import * as React from "react";
import { IBasicFormParam } from "shared/types";

export interface IStringParamProps {
  id: string;
  name: string;
  label: string;
  inputType?: string;
  param: IBasicFormParam;
  handleBasicFormParamChange: (
    name: string,
    p: IBasicFormParam,
  ) => (e: React.FormEvent<HTMLInputElement>) => void;
}

class TextParam extends React.Component<IStringParamProps> {
  public render() {
    const { id, name, param, label, inputType } = this.props;
    return (
      <div>
        <label htmlFor={id}>
          <div className="row">
            <div className="col-3 block">
              <div className="centered">{label}</div>
            </div>
            <div className="col-9 margin-t-small">
              <input
                id={id}
                onChange={this.props.handleBasicFormParamChange(name, param)}
                value={param.value === undefined ? "" : param.value}
                type={inputType ? inputType : "text"}
              />
              {param.description && <span className="description">{param.description}</span>}
            </div>
          </div>
        </label>
      </div>
    );
  }
}

export default TextParam;
