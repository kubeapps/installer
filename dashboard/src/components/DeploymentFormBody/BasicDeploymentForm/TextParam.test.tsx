import * as React from "react";

import { mount } from "enzyme";
import TextParam from "./TextParam";

const param = { path: "username", value: "user", type: "string" };
const defaultProps = {
  id: "foo",
  label: "Username",
  param,
  handleBasicFormParamChange: jest.fn(() => jest.fn()),
};

it("should render a text param with title and description", () => {
  const wrapper = mount(<TextParam {...defaultProps} />);
  const input = wrapper.find("input");
  expect(input.prop("value")).toBe(defaultProps.param.value);
  expect(wrapper).toMatchSnapshot();
});

it("should set the input type as number", () => {
  const wrapper = mount(<TextParam {...defaultProps} inputType={"number"} />);
  const input = wrapper.find("input");
  expect(input.prop("type")).toBe("number");
});

it("should forward the proper value", () => {
  const handler = jest.fn();
  const handleBasicFormParamChange = jest.fn(() => handler);
  const wrapper = mount(
    <TextParam {...defaultProps} handleBasicFormParamChange={handleBasicFormParamChange} />,
  );
  const input = wrapper.find("input");

  const event = { currentTarget: {} } as React.FormEvent<HTMLInputElement>;
  (input.prop("onChange") as any)(event);

  expect(handleBasicFormParamChange.mock.calls[0][0]).toEqual({
    path: "username",
    type: "string",
    value: "user",
  });
  expect(handler.mock.calls[0][0]).toMatchObject(event);
});

it("should set the input value as empty if the param value is not defined", () => {
  const tparam = { path: "username", type: "string" };
  const tprops = {
    id: "foo",
    name: "username",
    label: "Username",
    param: tparam,
    handleBasicFormParamChange: jest.fn(() => jest.fn()),
  };
  const wrapper = mount(<TextParam {...tprops} />);
  const input = wrapper.find("input");
  expect(input.prop("value")).toBe("");
});
