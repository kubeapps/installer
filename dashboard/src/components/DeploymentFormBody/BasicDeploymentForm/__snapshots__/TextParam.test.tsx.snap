// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a text param with title and description 1`] = `
<TextParam
  handleBasicFormParamChange={[Function]}
  id="foo"
  label="Username"
  param={
    Object {
      "path": "username",
      "type": "string",
      "value": "user",
    }
  }
>
  <div>
    <label
      htmlFor="foo"
    >
      <div
        className="row"
      >
        <div
          className="col-3 block"
        >
          <div
            className="centered"
          >
            Username
          </div>
        </div>
        <div
          className="col-9 margin-t-small"
        >
          <input
            id="foo"
            onChange={[Function]}
            type="text"
            value="user"
          />
        </div>
      </div>
    </label>
  </div>
</TextParam>
`;
